Content.Server/GameObjects/Components/Kitchen/KitchenMicrowaveComponent.cs

```diff
@@ -0,0 +1,389 @@
+        {
+            var itemEntity = eventArgs.User.GetComponent<HandsComponent>().GetActiveHand.Owner;
+
+            if(itemEntity.TryGetComponent<PourableComponent>(out var attackPourable))
+            {
+                //Get target and check if it can be poured into
+                if (!Owner.TryGetComponent<SolutionComponent>(out var mySolution)
+                    || !mySolution.CanPourIn)
+                {
+                    return false;
+                }
+
+                if (!itemEntity.TryGetComponent<SolutionComponent>(out var attackSolution)
+                    || !attackSolution.CanPourOut)
+                {
+                    return false;
+                }
+
+                //Get transfer amount. May be smaller than _transferAmount if not enough room
+                var realTransferAmount = ReagentUnit.Min(attackPourable.TransferAmount, mySolution.EmptyVolume);
+                if (realTransferAmount <= 0) //Special message if container is full
+                {
+                    _notifyManager.PopupMessage(Owner.Transform.GridPosition, eventArgs.User,
+                        Loc.GetString("Container is full"));
+                    return false;
+                }
+
+                //Move units from attackSolution to targetSolution
+                var removedSolution = attackSolution.SplitSolution(realTransferAmount);
+                if (!mySolution.TryAddSolution(removedSolution))
+                {
+                    return false;
+                }
+
+                _notifyManager.PopupMessage(Owner.Transform.GridPosition, eventArgs.User,
+                    Loc.GetString("Transferred {0}u", removedSolution.TotalVolume));
+                return true;
+            }
+
+            if (!itemEntity.TryGetComponent(typeof(FoodComponent), out var food))
+            {
+                return false;
+            }
+
+            var ent = food.Owner; //Get the entity of the ItemComponent.
+            _storage.Insert(ent);
+            UpdateUserInterface();
+            return true;
+
+        }
+
+        //This is required. It's 'cook'.
+        private void wzhzhzh()
+        {
+            _busy = true;
+            // Convert storage into Dictionary of ingredients
+            _solids.Clear();
+            foreach(var item in _storage.ContainedEntities)
+            {
+                if(_solids.ContainsKey(item.Prototype.ID))
+                {
+                    _solids[item.Prototype.ID]++;
+                }
+                else
+                {
+                    _solids.Add(item.Prototype.ID, 1);
+                }
+            }
+
+            // Check recipes
+            FoodRecipePrototype recipeToCook = null;
+            foreach(var r in _recipeManager.Recipes)
+            {
+                if (!CanSatisfyRecipe(r))
+                {
+                    continue;
+                }
+
+                recipeToCook = r;
```

Comment:
OK, we satisfy this recipe, which atm just means we have all the ingredients for that recipe in atleast the right quantities, what if there's a more complicated recipe that we match with, surely we'd want to go for that one? else we may end up in scenarios where certain complex recipes are uncookable because there's an easier recipe that the game keeps picking first (e.g, imagine a dish which contains fries, but there's also just a recipe to cook fries, the microwave would always cook the fries instead of making the other dish)

The entire loop should be replaced with something like this: 

```csharp
var cookableRecipes = _recipeManager.Recipes.Where(r => CanSatisfyRecipe(r));
cookableRecipes = cookableRecipes.Sort((a,b) => {
    if(a.IngredientsReagents.Count > b.IngredientsReagents.Count) return 1;
    if(a.IngredientsSolids.Count > b.IngredientsSolids.Count)     return 1;
    return -1;
});
if(cookableRecipes.Any()){
    recipeToCook = cookableRecipes.First();
}
```

This will give you the most complicated recipe you can do, or no recipe if you can't do any.
Which should prevent "can't cook dish X because it contains as a subset of its ingredients the same ingredients as dish Y"